src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs

```diff
@@ -51,11 +52,33 @@ internal static void GetDefaultIdentityAndHash(out object identity, out string h
             Assembly? assembly = Assembly.GetEntryAssembly();
 
             if (assembly == null)
+            {
                 throw new IsolatedStorageException(SR.IsolatedStorage_Init);
+            }
+
+            // in case of SingleFile, Location.Length is 0, we will use native entrypoint's filename.
+            if (assembly.Location.Length == 0)
+            {
+                using (Process currentProcess = Process.GetCurrentProcess())
```

Comment:
We shouldn’t be adding a dependency on Process just to get process name.  Certainly we should be able to use a lower level API in AppContext, Reflection, or AppDomain to get this? @swaroop-sridhar 

Also we should move this down into a fallback case below so we don’t change the order of precedence.

Reply:
Perhaps simply moving codebase under else check might fix 🤔

Reply:
>  Certainly we should be able to use a lower level API in AppContext, Reflection, or AppDomain to get this?

There is no simple public API to get process name today. https://github.com/dotnet/runtime/issues/40862 is a proposal to add it.

> @swaroop-sridhar

Swaroop changed teams recently. He does not work on this anymore.

Reply:
AFAIK, currently there is no better alternative than using S.D.Process (that's a general guideline for 5.0 I have found on multiple SingleFile related issues from @swaroop-sridhar). However, in 6.0 we will potentially get `Environment.ProcessPath` (https://github.com/dotnet/runtime/issues/40862 - API approved for 6.0) and this can be replaced then.

Reply:
Just refreshed the page and saw comment by @jkotas. :)

Reply:
I see.  Well we can at least go down the strong-name path below without accessing `Assembly.Codebase`, move the `Codebase` access under the else.  Then we need to think of an alternative to codebase for non-strong-named assemblies.

Would it make sense to try `AppContext.BaseDirectory` + assembly.Name instead?   This is close to equivalent (minus extension) with what would have been used in 3.1.  I see in 3.1 each publish of the app would get a unique path under `%TEMP%\.net\singleFile\<app>\` so we don't need to be concerned with getting the *same* value here since app would need to rebuild to run on 5.0 and thus would get a new path here.

Reply:
Makes sense. I have removed the dependency on S.D.Process and to avoid dependency on RuntimeInformation, defined constant ExeFileExtension in Helper.{Unix,Win32}.cs files. It seem to have covered the strong-name/non-strong-name SingleFile matrix.